### DISCUSSION TITLE
HELP Push Modal Page Not Working! Migrate from Xamarin.Forms to MAUI Prism Library

### DISCUSSION THREAD

**Discussion by giuseppenovielli** [PrismLibrary/Prism#2828] [.NET MAUI]
Hi,
thanks @dansiegel @brianlagunas  and all contributors for release Prism.MAUI!!

I have migrate code from Xamarin.Forms to MAUI.
--> https://github.com/giuseppenovielli/MAUI_XF-Prism-Full-Navigation-Example

_You can clone this repository and execute maui and xamarin.forms app to show differents.
The code is EQUAL_ 

With new Prism.Maui, some Prism.Xamarin.Forms API are not found!

**With Xamarin.Forms if i want to push a modal page, i write**

```
private async Task PageModalClick()
{
    var navResult = await NavigationService.NavigateAsync(nameof(NavigationPage) + "/" + nameof(Page1ModalPage), useModalNavigation: true);
}
```
https://github.com/giuseppenovielli/MAUI_XF-Prism-Full-Navigation-Example/blob/master/PrismFullNavigation/xamarin_forms_5/PrismFullNavigation/ViewModels/MainPageViewModel.cs#L61

**With MAUI:**
`useModalNavigation: true ` --> is not found!

I'm try to reproduce ModalPage but without success!

```
private async Task PageModalClick()
        {
            var navParameters = new NavigationParameters();
            navParameters.Add(KnownNavigationParameters.UseModalNavigation, true);
            var navResult = await NavigationService.NavigateAsync(nameof(NavigationPage) + "/" + nameof(Page1ModalPage), navParameters);
        }
```

OR

```
private async Task PageModalClick()
{
  var result = await NavigationService.CreateBuilder()
                    .AddNavigationPage(true)
                    .AddSegment(nameof(Page1ModalPage), s => s.UseModalNavigation())
                    .NavigateAsync();
 }
```

https://github.com/giuseppenovielli/MAUI_XF-Prism-Full-Navigation-Example/blob/master/PrismFullNavigation/maui_net7/PrismFullNavigation/ViewModels/MainPageViewModel.cs#L61

The MAUI's code push new page, always from current navigation page, this is wrong if i want a ModalPush.

**What am I doing wrong?
I want the same ModalPage like Xamarin.Forms.**

Thanks!

**Comment by mnxamdev** [comment]
Agreed, there seems to be an issue with showing a page as modal in Prism Maui. The page ends up always getting shoved into the flyout detail page and not shown as a modal. I pulled a sample project that showcase modals but it was referencing an older version of Prism Maui source and modals worked. I then changed it to use latest Prism.Maui nuget and modals stopped working and pages are loaded in the flyout detail. @giuseppenovielli did you create a github issue for this yet?

**Comment by giuseppenovielli** [comment]
@mnxamdev which prism.maui's version work with ModalPage?

**Reply by mnxamdev** [reply]
Whatever version this developer was using in his zip package. I seem to recall downloading and running this but it looks like the example of it NOT working. However, if you play around with the example and press "Flyout Page" then open the menu and press "Root Page" then press "Show Modal" it works. However, I don't think that is desirable because when you press "Flyout" then "Root" you don't have a flyout menu anymore. 
https://github.com/PrismLibrary/Prism.Maui/issues/106#issuecomment-1225866058

Note I believe there is a github issue for this as well.
https://github.com/PrismLibrary/Prism/issues/2819

**Comment by giuseppenovielli** [comment]
@dansiegel @brianlagunas when possible can you check if this is a bug or we write a wrong code, to show modal pages?

Thanks!

**Comment by eduardo-juarez-dev** [comment]
@dansiegel @brianlagunas This is still an issue at the moment. Have tried everything but can't get a page to open modally.
I even tried to add "?useModalNavigation=true" at the end of the NavigateAsync call. Are we missing something?

currently using: prism.maui 8.1.273-pre

**Comment by giuseppenovielli** [comment]
Any news? @brianlagunas @dansiegel

**Reply by soroshsabz** [reply]
ITNOA

I think MAUI version is not update for one years, and this is very strange

https://github.com/PrismLibrary/Prism/discussions/2870

**Reply by brianlagunas** [reply]
> ITNOA
> 
> I think MAUI version is not update for one years, and this is very strange
> 
> #2870

Our GitHub sponsors have access to updated versions of our MAUI builds. Everyone else will get the updated packages with the release of .NET 8.

**Reply by soroshsabz** [reply]
thanks for response

**Reply by robertqtk** [reply]
> > ITNOA
> > I think MAUI version is not update for one years, and this is very strange
> > #2870
> 
> Our GitHub sponsors have access to updated versions of our MAUI builds. Everyone else will get the updated packages with the release of .NET 8.

@brianlagunas 

Then it wouldn't be a stretch to say it is in best interest to not create production apps in Prism.Maui until the release of .NET 8? We have a project starting soon and I was hoping we could dive right into Prism.Maui. If the library needs more time to mature, then perhaps we'll leverage the community toolkit MVVM structure for now. Or, we start the project in xamarin.forms to be updated upon release of .NET 8. In you honest opinion, what do you think would be easier to convert to Prism.Maui?

Appreciate all you guys do for the community!

**Reply by dansiegel** [reply]
It's more fair to say that .NET MAUI needs more time to mature

**Reply by robertqtk** [reply]
Hey Dan,

I agree with you. With that statement in mind, would you start a production app (mobile) in Maui in the current ecosystem? Or start in Xamarin and migrate to Maui when the dust settles? 

Thanks for the feedback

**Comment by giuseppenovielli** [comment]
Hi guys,

Modal Page works with following instructions:

Packages:
https://github.com/dotnet/maui/releases/tag/8.0.3
https://www.nuget.org/packages/Prism.DryIoc.Maui/9.0.271-pre

Platforms
Android -> Work
iOS -> Not tested yet
MacCatalyst -> Not tested yet
Windows -> Can't test

Code:

```
var result = await NavigationService.CreateBuilder()
                    .AddNavigationPage(true)
                    .AddSegment(nameof(<namepage>), s => s.UseModalNavigation())
                    .NavigateAsync();
```

Thanks @dansiegel @brianlagunas !!

Soon upgrade my project -> https://github.com/giuseppenovielli/MAUI_XF-Prism-Full-Navigation-Example
